context/app/routes_cells.py

```diff
@@ -217,13 +220,15 @@ def _get_matched_cell_counts_per_cluster(cells):
                            'modality': 'Z',
                            'unmatched': 0}]}
     '''
-    group_keys = ["cluster_name", "cluster_number", 'modality']
+    group_keys = ["cluster_name", "cluster_number"]
+    if query_type == 'gene':
+        group_keys.append('modality')
```

Comment:
```suggestion
    group_keys = ["cluster_name", "cluster_number", 'modality']
```
With the suggestion above, `modality` will be present, though it may be `None`.